src/config_file.c

```diff
@@ -1041,8 +1041,9 @@ static int parse_section_header_ext(struct reader *reader, const char *line, con
 	GITERR_CHECK_ALLOC_ADD(&alloc_len, base_name_len, quoted_len);
 	GITERR_CHECK_ALLOC_ADD(&alloc_len, alloc_len, 2);
 
-	git_buf_grow(&buf, alloc_len);
-	git_buf_printf(&buf, "%s.", base_name);
+	if (git_buf_grow(&buf, alloc_len) < 0
+	    || git_buf_printf(&buf, "%s.", base_name) < 0)
+		goto end_parse;
```

Comment:
I'm a bit torn, as well. Anyway, I think failing early here is a nice to have and it shuts up Coverity without any real detriment.

Huh. Where did I take up that "||" should start at newline... dunno. Will fix all occasions, thanks.